src/dynatrace/opentelemetry/metrics/export/_factory.py

```diff
@@ -152,3 +153,23 @@ def _log_temporality_mismatch(
                               metric.name,
                               metric.data.aggregation_temporality.name,
                               supported_temporality.name)
+
+    def _retain_string_valued_dimensions(
+            self,
+            attributes: Optional[Mapping]) -> Optional[Mapping[str, str]]:
+
+        if not attributes:
+            return attributes
+
+        return dict(
+            filter(lambda attr: self._is_string_value(attr[1]),
```

Comment:
this will drop non-string values but retain non-string keys, right? :thinking:

Reply:
yes, it iterates tuples of (key, value), thus the `attr[1]` refers to the value.

Reply:
I think we might also want to drop non-string keys as we'll run into the same problem there right? :thinking:

Reply:
True, what a nice thing with not statically typed languages, right? ✨